core/templates/pages/exploration-editor-page/services/exploration-data.service.ts

```diff
@@ -149,7 +150,9 @@ export class ExplorationDataService {
             this.explorationId).then((response) => {
             this.loggerService.info('Retrieved exploration data.');
             this.loggerService.info(JSON.stringify(response));
-            this.draftChangeListId = response.draft_change_list_id;
+            if (response.draft_change_list_id) {
+              this.draftChangeListId = response.draft_change_list_id;
+            }
```

Comment:
Why is this `if` needed?

Reply:
As draft changed list id is an optional parameter, it may be possible it is undefined so passing undefines was giving type error and not needed.

Reply:
I tried looking for this and this is what I found out (I may be wrong) whenever we call this function getDataAsync, it first tries to find the exploration data saved in the backend, if that is present (then draft_change_list_id will be undefined). Otherwise, it will return the saved draft exploration where this property is defined. But to remove the type error we need to add the 'if' condition, so that compiler is happy that this is not undefined.

Reply:
Hmm, I'm looking into the backend code https://github.com/oppia/oppia/blob/develop/core%2Fdomain%2Fexp_services.py#L1867-L1878 sand here we always set `draft_change_list_id` to some value, either the ID or zero. So I think that here https://github.com/oppia/oppia/blob/develop/core%2Ftemplates%2Fdomain%2Fexploration%2FExplorationObjectFactory.ts#L62 we don't need to make `draft_change_list_id` optional as it will always be set.

Reply:
Done. Thanks!